spec/lib/food_robot_spec.rb

```diff
@@ -0,0 +1,294 @@
+require 'rails_helper'
+
+RSpec.describe FoodRobot do
+  let(:monday) { Time.zone.today.monday }
+  let(:tuesday) { monday + 1.day }
+
+  describe '::generate_log_entries' do
+    context 'without an absence' do
+      context 'with an irregular schedule chain' do
+        let!(:chain) { create(:schedule_chain, day_of_week: monday.wday, irregular: true) }
+        let!(:donor) { create(:donation_schedule, schedule_chain: chain, position: 1) }
+        let!(:recipient) { create(:recipient_schedule, schedule_chain: chain, position: 2) }
+
+        subject { described_class.generate_log_entries(monday) }
+
+        it 'generates no logs' do
+          expect { subject }.not_to change { Log.count }
+        end
+      end
+
+      context 'with a non-functional schedule chain' do
```

Comment:
Dunno if it's possible to create an empty chain or not, but might be worth spec'ing that case too? Since it seems like we're generating a lot of these things.